commit cdc6c4e518c3944b716050aea18c3698bc30a7a2
Author: Lily Foster <lily@lily.flowers>
Date:   Wed Mar 16 11:10:10 2022 -0400

    add mp4 cover art support

diff --git a/src/mpDris2.in.py b/src/mpDris2.in.py
index 4719dda..3fe3cc2 100755
--- a/src/mpDris2.in.py
+++ b/src/mpDris2.in.py
@@ -665,6 +665,16 @@ class MPDWrapper(object):
                         if pic.type == mutagen.id3.PictureType.COVER_FRONT:
                             self._temp_song_url = song_url
                             return self._create_temp_cover(pic)
+                elif song.tags and "covr" in song.tags:
+                    # MP4
+                    for data in song.get("covr", []):
+                        mimes = {mutagen.mp4.AtomDataType.JPEG: "image/jpeg",
+                                 mutagen.mp4.AtomDataType.PNG: "image/png"}
+
+                        pic = mutagen.id3.APIC(mime=mimes.get(data.imageformat, ""), data=data)
+
+                        self._temp_song_url = song_url
+                        return self._create_temp_cover(pic)
 
             # Look in song directory for common album cover files
             if os.path.exists(song_dir) and os.path.isdir(song_dir):
