diff --git a/app/server/ruby/bin/daemon.rb b/app/server/ruby/bin/daemon.rb
index 0809025f7..09c02e40b 100644
--- a/app/server/ruby/bin/daemon.rb
+++ b/app/server/ruby/bin/daemon.rb
@@ -614,7 +614,7 @@ module SonicPi
         case Util.os
         when :linux, :raspberry
           #Start Jack if not already running
-          if `ps cax | grep jackd`.split(" ").first.nil?
+          if `jack_wait -c`.include? 'not running'
             #Jack not running - start a new instance
             Util.log "Jackd not running on system. Starting..."
             @jack_booter = JackBooter.new
