diff --git a/app/external/CMakeLists.txt b/app/external/CMakeLists.txt
index f918b76ba..3652a699d 100644
--- a/app/external/CMakeLists.txt
+++ b/app/external/CMakeLists.txt
@@ -8,7 +8,7 @@ project(AubioBuilder
     VERSION 1.0.0
     )
 
-
+  include(GNUInstallDirs)
   include(ExternalProject)
 
   set(CMAKE_OSX_DEPLOYMENT_TARGET '10.14')
@@ -108,15 +108,15 @@ ExternalProject_Add(libsndfile
         -DCMAKE_POSITION_INDEPENDENT_CODE=ON
         -DFLAC_ROOT=${CMAKE_BINARY_DIR}/flac-package
         -DPC_OGG_INCLUDE_DIRS=${CMAKE_BINARY_DIR}/ogg-package/include
-        -DPC_OGG_LIBRARY_DIRS=${CMAKE_BINARY_DIR}/ogg-package/lib
+        -DPC_OGG_LIBRARY_DIRS=${CMAKE_BINARY_DIR}/ogg-package/${CMAKE_INSTALL_LIBDIR}
         -DPC_FLAC_INCLUDE_DIRS=${CMAKE_BINARY_DIR}/flac-package/include
-        -DPC_FLAC_LIBRARY_DIRS=${CMAKE_BINARY_DIR}/flac-package/lib
+        -DPC_FLAC_LIBRARY_DIRS=${CMAKE_BINARY_DIR}/flac-package/${CMAKE_INSTALL_LIBDIR}
         -DPC_VORBIS_INCLUDE_DIRS=${CMAKE_BINARY_DIR}/vorbis-package/include
-        -DPC_VORBIS_LIBRARY_DIRS=${CMAKE_BINARY_DIR}/vorbis-package/lib
+        -DPC_VORBIS_LIBRARY_DIRS=${CMAKE_BINARY_DIR}/vorbis-package/${CMAKE_INSTALL_LIBDIR}
         -DPC_VORBISENC_INCLUDE_DIRS=${CMAKE_BINARY_DIR}/vorbis-package/include
-        -DPC_VORBISENC_LIBRARY_DIRS=${CMAKE_BINARY_DIR}/vorbis-package/lib
+        -DPC_VORBISENC_LIBRARY_DIRS=${CMAKE_BINARY_DIR}/vorbis-package/${CMAKE_INSTALL_LIBDIR}
         -DPC_OPUS_INCLUDE_DIRS=${CMAKE_BINARY_DIR}/opus-package/include
-        -DPC_OPUS_LIBRARY_DIRS=${CMAKE_BINARY_DIR}/opus-package/lib
+        -DPC_OPUS_LIBRARY_DIRS=${CMAKE_BINARY_DIR}/opus-package/${CMAKE_INSTALL_LIBDIR}
         -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}
 
         INSTALL_DIR ${CMAKE_BINARY_DIR}/libsndfile-package
@@ -130,11 +130,11 @@ ExternalProject_Add(aubio
 
     CMAKE_ARGS
         -DLIBSNDFILE_INCLUDE_DIR=${CMAKE_BINARY_DIR}/libsndfile-package/include
-        -DLIBSNDFILE_LIBRARY_DIR=${CMAKE_BINARY_DIR}/libsndfile-package/lib
-        -DLIBOGG_LIBRARY_DIR=${CMAKE_BINARY_DIR}/ogg-package/lib
-        -DLIBVORBIS_LIBRARY_DIR=${CMAKE_BINARY_DIR}/vorbis-package/lib
-        -DLIBOPUS_LIBRARY_DIR=${CMAKE_BINARY_DIR}/opus-package/lib
-        -DLIBFLAC_LIBRARY_DIR=${CMAKE_BINARY_DIR}/flac-package/lib
+        -DLIBSNDFILE_LIBRARY_DIR=${CMAKE_BINARY_DIR}/libsndfile-package/${CMAKE_INSTALL_LIBDIR}
+        -DLIBOGG_LIBRARY_DIR=${CMAKE_BINARY_DIR}/ogg-package/${CMAKE_INSTALL_LIBDIR}
+        -DLIBVORBIS_LIBRARY_DIR=${CMAKE_BINARY_DIR}/vorbis-package/${CMAKE_INSTALL_LIBDIR}
+        -DLIBOPUS_LIBRARY_DIR=${CMAKE_BINARY_DIR}/opus-package/${CMAKE_INSTALL_LIBDIR}
+        -DLIBFLAC_LIBRARY_DIR=${CMAKE_BINARY_DIR}/flac-package/${CMAKE_INSTALL_LIBDIR}
 
         -DPC_OGG_INCLUDE_DIRS=${CMAKE_BINARY_DIR}/ogg-package/include
         -DPC_VORBIS_INCLUDE_DIRS=${CMAKE_BINARY_DIR}/vorbis-package/include
