diff --git a/app/server/beam/print_erlang_version b/app/server/beam/print_erlang_version
old mode 100644
new mode 100755
diff --git a/app/server/beam/tau/boot-lin.sh b/app/server/beam/tau/boot-lin.sh
old mode 100644
new mode 100755
index afce0ed88..38788117a
--- a/app/server/beam/tau/boot-lin.sh
+++ b/app/server/beam/tau/boot-lin.sh
@@ -3,4 +3,4 @@ set -e # Quit script on error
 SCRIPT_DIR="$(dirname "$0")"
 cd ${SCRIPT_DIR}
 echo "Booting Sonic Pi on Linux..."
-TAU_ENABLED=$1 TAU_INTERNAL=$2 TAU_MIDI_ENABLED=$3 TAU_LINK_ENABLED=$4 TAU_IN_PORT=$5 TAU_API_PORT=$6 TAU_SPIDER_PORT=$7 TAU_DAEMON_PORT=$8 _build/"${MIX_ENV:-prod}"/rel/tau/bin/tau start
+TAU_ENABLED=$1 TAU_INTERNAL=$2 TAU_MIDI_ENABLED=$3 TAU_LINK_ENABLED=$4 TAU_IN_PORT=$5 TAU_API_PORT=$6 TAU_SPIDER_PORT=$7 TAU_DAEMON_PORT=$8 TAU_LOG_PATH=$9 _build/"${MIX_ENV:-prod}"/rel/tau/bin/tau start
diff --git a/app/server/beam/tau/boot-mac.sh b/app/server/beam/tau/boot-mac.sh
old mode 100644
new mode 100755
index d0da553d0..1e40cca6c
--- a/app/server/beam/tau/boot-mac.sh
+++ b/app/server/beam/tau/boot-mac.sh
@@ -3,4 +3,4 @@ set -e # Quit script on error
 SCRIPT_DIR="$(dirname "$0")"
 cd ${SCRIPT_DIR}
 echo "Booting Sonic Pi on macOS..."
-TAU_ENABLED=$1 TAU_INTERNAL=$2 TAU_MIDI_ENABLED=$3 TAU_LINK_ENABLED=$4 TAU_IN_PORT=$5 TAU_API_PORT=$6 TAU_SPIDER_PORT=$7 TAU_DAEMON_PORT=$8 _build/"${MIX_ENV:-prod}"/rel/tau/bin/tau start
+TAU_ENABLED=$1 TAU_INTERNAL=$2 TAU_MIDI_ENABLED=$3 TAU_LINK_ENABLED=$4 TAU_IN_PORT=$5 TAU_API_PORT=$6 TAU_SPIDER_PORT=$7 TAU_DAEMON_PORT=$8 TAU_LOG_PATH=$9 _build/"${MIX_ENV:-prod}"/rel/tau/bin/tau start
