diff --git a/app/server/beam/tau/boot-lin.sh b/app/server/beam/tau/boot-lin.sh
index 7ac4952dd..af32335b3 100644
--- a/app/server/beam/tau/boot-lin.sh
+++ b/app/server/beam/tau/boot-lin.sh
@@ -25,5 +25,5 @@ if [ $TAU_ENV == "dev" ]
 then
   mix run --no-halt > log/tau_stdout.log 2>&1
 else
-  _build/prod/rel/tau/bin/tau start > log/tau_stdout.log 2>&1
+  _build/prod/rel/tau/bin/tau start > /dev/null 2>&1
 fi
diff --git a/app/server/beam/tau/boot-mac.sh b/app/server/beam/tau/boot-mac.sh
index 27bd3701c..f48bbf04c 100755
--- a/app/server/beam/tau/boot-mac.sh
+++ b/app/server/beam/tau/boot-mac.sh
@@ -25,5 +25,5 @@ if [ $TAU_ENV == "dev" ]
 then
   mix run --no-halt > log/tau_stdout.log 2>&1
 else
-  _build/prod/rel/tau/bin/tau start > log/tau_stdout.log 2>&1
+  _build/prod/rel/tau/bin/tau start > /dev/null 2>&1
 fi
diff --git a/app/server/beam/tau/boot-win.bat b/app/server/beam/tau/boot-win.bat
index 6f3e59ab6..11d988162 100644
--- a/app/server/beam/tau/boot-win.bat
+++ b/app/server/beam/tau/boot-win.bat
@@ -30,5 +30,5 @@ set MIX_ENV=%TAU_ENV%
 IF "%TAU_ENV%" == "dev" (
   mix run --no-halt > log\tau_stdout.log 2>&1
 ) ELSE (
-  _build\prod\rel\tau\bin\tau start > log\tau_stdout.log 2>&1
+  _build\prod\rel\tau\bin\tau start > /dev/null 2>&1
 )
