diff --git a/tests/conftest.py b/tests/conftest.py
index 970c353..6d58378 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -6,7 +6,6 @@ import time
 import typing
 
 import pytest
-import trustme
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives.serialization import (
     BestAvailableEncryption,
@@ -20,6 +19,16 @@ from uvicorn.main import Server
 from httpx import URL
 from tests.concurrency import sleep
 
+try:
+    import trustme
+
+    # Ensure CA is available in runtime
+    trustme.CA()
+
+    TRUSTME: bool = True
+except ImportError:
+    TRUSTME = False
+
 ENVIRONMENT_VARIABLES = {
     "SSL_CERT_FILE",
     "SSL_CERT_DIR",
@@ -200,6 +209,8 @@ async def redirect_301(scope, receive, send):
 
 @pytest.fixture(scope="session")
 def cert_authority():
+    if not TRUSTME:
+        pytest.xfail("trustme is not supported")
     return trustme.CA()
 
 
